server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -265,6 +266,23 @@ public CopyTo copyTo() {
         return copyTo;
     }
 
+    private String readComplexJsonElement(XContentParser parser, int tokenDepth) throws IOException {
+        XContentParser.Token closingToken = null;
+        if (parser.currentToken() == XContentParser.Token.START_ARRAY) {
+            closingToken = XContentParser.Token.END_ARRAY;
+        } else if (parser.currentToken() == XContentParser.Token.START_OBJECT) {
+            closingToken = XContentParser.Token.END_OBJECT;
+        } else throw new IOException("Incorrect token type");
```

Comment:
As it is not really an exception, I would suggest instead return `null`

Reply:
and then check if we return `null` instead of handling `innerException`?